### DISCUSSION TITLE
Resolving tables with key across sessions

### DISCUSSION THREAD

**Discussion by nacjacks** [scandum/tintin#177] [General]
I have two sessions running, each one reads a table in as a class that contains character buff data. When I iterate over the data in one session I am trying to check the status from the other session and coming up with empty when using the key but using the actual string key I get the status.

Any ideas how to grab the status from the sec session?

```
#nop buff data class

#CLASS {tracked_buffs} OPEN

#VARIABLE {tracked_buffs} {
    {Barkskin}
    {
        {class}{dr} {display_text}{Barkskin} {status}{off} {buff_type}{1} {cast_command}{cast 'barkskin'}
        {buff_level} {1}
        {character_with_buff} {Vespa;}
    }

#CLASS {tracked_buffs} CLOSE
```

```
#nop loop
#FOREACH {*tracked_buffs[%*]} {key} {
#SHOWME {$key}; #nop OUTPUT Barkskin
#SHOWME {$tracked_buffs[$key][status]};#nop OUTPUT on
#SHOWME {@sec{$tracked_buffs[$key][status]}};#nop OUTPUT blank
#SHOWME {@sec{$tracked_buffs[Barkskin][status]}};#nop OUTPUT on
}
```

**Comment by scandum** [comment]
This should work. You could also use an alias and force substitution through the alias arguments before calling @sec.
```
#line sub var #SHOWME {@sec{$$tracked_buffs[$key][status]}};
```
